lightning/src/blinded_path/payment.rs

```diff
@@ -92,6 +99,27 @@ enum BlindedPaymentTlvsRef<'a> {
 	Receive(&'a ReceiveTlvs),
 }
 
+impl<'a> BlindedPaymentTlvsRef<'a> {
+	pub(crate) fn pad_tlvs(self, max_length: usize) -> BlindedPaymentTlvs {
+		let length = max_length.checked_sub(self.serialized_length());
+		debug_assert!(length.is_some(), "Size of this packet should not be larger than the size of the largest packet.");
+		let padding = Some(Padding::new(length.unwrap()));
+
+		match self {
+			BlindedPaymentTlvsRef::Forward(tlvs) => {
+				let mut new_tlvs = tlvs.clone();
```

Comment:
We shouldn't clone the underlying TLVs. We'll want this method to return `BlindedPaymentTlvsRef` instead of `BlindedPaymentTlvs`.

Reply:
As per our offline discussion, shifted to using the `BlindedPaymentTlvs` directly to allow mutating the underlying tlvs, instead of updating the `BlindedPaymentTlvsRef` in [pr3177.09](https://github.com/shaavan/rust-lightning/commits/pr3177.09)

Thanks a lot for all the suggestions and feedback!